### DISCUSSION TITLE
Looping streaming songs without buffering again ?

### DISCUSSION THREAD

**Discussion by veganomy** [mpv-player/mpv#15451] [Q&A] [answered]
When I loop a song from youtube music, mpv always reloads it again and again instead of reusing the already buffered song. Is there anyway to fix this ?  

I know I can download and loop them. Just asking if there's a way without downloading and reusing the same buffer while streaming.

**Comment by guidocella** [comment]
Looping doesn't redownload if the whole file is cached. Try to increase `--demuxer-max-bytes`.

**Reply by veganomy** [reply]
Can `cache-secs` be used instead of this ? Caz that's certain for me. Size is not certain, as the bitrate varies a lot.

**Reply by guidocella** [reply]
Used for what? `--cache-secs` is basically unlimited by default so it can't cause rebuffering when looping unless you go out of your way to configure it.

**Reply by veganomy** [reply]
I did set cache to yes and `demuxer-max-bytes` to like 100MiB. The stream after ending, still seems to rebuffer again. Anything I'm missing ?

Btw, I use `loop-playlist=inf` caz a single stream is also considered a playlist. Does the issue has something to do with playlist cache ?

**Reply by guidocella** [reply, marked as answer]
Yes, you need to use `--loop-file`, this is documented in https://mpv.io/manual/master/#options-loop-file

**Reply by veganomy** [reply]
Ok seems to work now. Is there anyway to make it work for `loop-playlist` too ? looping a file seems to not proceed to next song unless I explicitly `next` it. Especially when I wanna listen to an album in repeat.

**Reply by guidocella** [reply]
```
[loop-single]
profile-cond=playlist_count == 1
loop-file
```

**Reply by veganomy** [reply]
I've understood the problem now. As soon as the last file playback is left, mpv seems to discard the cache and proceed with the next one. This is why `loop-playlist` didn't work.

I've tried the profile condition too. Didn't seem to work for looping playlist without discarding cache.
